cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -100,6 +154,63 @@ void contingencyMatrix(const T* groundTruth,
                                      maxLabel);
 }
 
+/**
+ * @brief contruct contingency matrix given input ground truth and prediction
+ *        labels. Users should call function getInputClassCardinality to find
+ *        and allocate memory for output. Similarly workspace requirements
+ *        should be checked using function getContingencyMatrixWorkspaceSize
+ * @tparam DataT label type
+ * @tparam OutType output matrix type
+ * @tparam IdxType Index type of matrix extent.
+ * @tparam LayoutPolicy Layout type of the input data.
+ * @param handle: the raft handle.
+ * @param groundTruth: device 1-d array for ground truth (num of rows)
+ * @param predictedLabel: device 1-d array for prediction (num of columns)
+ * @param outMat: output buffer for contingency matrix
+ * @param workspace: Optional, workspace memory allocation
+ * @param minLabel: Optional, min value in input ground truth array
+ * @param maxLabel: Optional, max value in input ground truth array
+ */
+template <typename DataT,
+          typename OutType,
+          typename IdxType,
+          typename LayoutPolicy,
+          typename WorkspaceDataType>
+void contingencyMatrix(
```

Comment:
Use snake case everywhere for consistency (eg `contingency_matrix`). This applies to function names, arguments. Also templates if there's time but we can fix those later if needed.